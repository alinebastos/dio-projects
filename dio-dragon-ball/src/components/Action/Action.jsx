import React, { useState } from 'react';
import {
  Container,
  Row,
  Col,
  Button,
  Modal,
  ModalFooter,
  ModalBody,
} from 'reactstrap';
import { Text } from 'rebass';
import styled from 'styled-components';

const CardContent = styled.div`
  background: #fafafa;
  text-align: center;
  display: flex;
  align-items: center;
  justify-content: center;
  height: 150px;
  flex-direction: column;
`;

const Shenlong = styled.img`
  height: 400px;
  position: fixed;
  z-index: 9;
  left: 40%;
  width: auto;
`;

const Action = ({ balls }) => {
  const [modal, setModal] = useState(false);
  const [show, setShow] = useState(false);

  const toggle = () => setModal(!modal);

  const invoke = () => {
    !!(balls?.length === 7) ? setShow(true) : setModal(true);
  };

  return (
    <>
      {show && (
        <Shenlong data-testid="shenlong" src='https://static.wixstatic.com/media/ecfb5e_dbd3be9fe8f14cc2a37e91167d125bc2.png/v1/fill/w_784,h_784,al_c,q_95,usm_0.66_1.00_0.01/ecfb5e_dbd3be9fe8f14cc2a37e91167d125bc2.webp' />
      )}
      <CardContent data-testid="card-shenlong">
        <Text fontSize='32px'>Invocar shenlong</Text>
        <Button color='warning' onClick={invoke} data-testid="invoke-button">
          Invocar
        </Button>
      </CardContent>
      <Modal isOpen={modal} toggle={toggle}>
        <Text data-testid="modaltext">Você não tem todas as esferas para invocar o shenlong</Text>
        <ModalFooter>
          <Button color='secondary' data-testid="back" onClick={toggle}>
            Voltar
          </Button>
        </ModalFooter>
      </Modal>
    </>
  );
};

export default Action;
