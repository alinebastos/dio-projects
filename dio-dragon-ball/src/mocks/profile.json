{
  "profile": {
    "id": 1,
    "name": "La√≠s Lima",
    "image": "https://pbs.twimg.com/profile_images/1252965332785991680/8Ry7z6Pt_400x400.jpg",
    "balls": [1, 5, 6, 7]
  }
}
