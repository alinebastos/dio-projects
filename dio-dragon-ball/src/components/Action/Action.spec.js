import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Action from './Action'

//mocks
import { profile } from '../../mocks/profile.json';
import profileSuccess from '../../mocks/profileSuccess.json';

it('Should render Action', () => {
  const { container, debug, getByText } = render(<Action balls={profile.balls} />)
  const button = getByText('Invocar').closest('button')

  expect(getByText('Invocar shenlong')).toBeInTheDocument()
  expect(button).toBeInTheDocument()
  expect(container).toMatchSnapshot()
})

it('Should open modal if user dont have all dragon balls', () => {
  const { debug, getByText } = render(<Action balls={profile.balls} />)
  const button = getByText('Invocar').closest('button')

  fireEvent.click(button)
  expect(getByText('Você não tem todas as esferas para invocar o shenlong')).toBeInTheDocument()

  const backButton = getByText('Voltar').closest('button')
  fireEvent.click(backButton)

  expect(getByText('Invocar shenlong')).toBeInTheDocument()
})

it('Should show shenlong', () => {
  const { container, debug, getByText, getByTestId } = render(<Action balls={profileSuccess.profile.balls} />)
  const button = getByText('Invocar').closest('button')

  fireEvent.click(button)
  expect(getByTestId('shenlong')).toBeInTheDocument()

  expect(container).toMatchSnapshot()
})


