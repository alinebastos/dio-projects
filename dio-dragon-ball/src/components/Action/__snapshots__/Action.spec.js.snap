// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Action 1`] = `
<div>
  <div
    class="sc-AxjAm gqoJub"
    data-testid="card-shenlong"
  >
    <div
      class="css-1brk784"
    >
      Invocar shenlong
    </div>
    <button
      class="btn btn-warning"
      data-testid="invoke-button"
      type="button"
    >
      Invocar
    </button>
  </div>
</div>
`;

exports[`Should show shenlong 1`] = `
<div>
  <img
    class="sc-AxirZ hyvxts"
    data-testid="shenlong"
    src="https://static.wixstatic.com/media/ecfb5e_dbd3be9fe8f14cc2a37e91167d125bc2.png/v1/fill/w_784,h_784,al_c,q_95,usm_0.66_1.00_0.01/ecfb5e_dbd3be9fe8f14cc2a37e91167d125bc2.webp"
  />
  <div
    class="sc-AxjAm gqoJub"
    data-testid="card-shenlong"
  >
    <div
      class="css-1brk784"
    >
      Invocar shenlong
    </div>
    <button
      class="btn btn-warning"
      data-testid="invoke-button"
      type="button"
    >
      Invocar
    </button>
  </div>
</div>
`;
